src/System.Xml.XmlSerializer/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -0,0 +1,1242 @@
+                    if ((readOnly && o == null) || Reader.IsEmptyElement)
+                    {
+                        Reader.Skip();
+                    }
+                    else
+                    {
+                        Reader.ReadStartElement();
+                        Reader.MoveToContent();
+                        UnknownNodeAction unknownNode = UnknownNodeAction.ReadUnknownNode;
+                        while (Reader.NodeType != XmlNodeType.EndElement && Reader.NodeType != XmlNodeType.None)
+                        {
+                            Member temp;
+                            WriteMemberElements(ref o, collectionMember, out temp, new Member[] { new Member(memberMapping) }, unknownNode, unknownNode, null, null);
+                            Reader.MoveToContent();
+                        }
+                        ReadEndElement();
+                    }
+
+                    SetCollectionObjectWithCollectionMember(ref o, collectionMember, collectionType);
+
+                }
+            }
+        }
+
+        private object WritePrimitive(TypeMapping mapping, Func<string> readFunc)
+        {
+            if (mapping is EnumMapping)
+            {
+                if (mapping.IsSoap)
+                {
+                    throw new PlatformNotSupportedException("mapping.IsSoap");
+                }
+
+                return WriteEnumMethod((EnumMapping)mapping, readFunc);
+            }
+            else if (mapping.TypeDesc == StringTypeDesc)
+            {
+                return readFunc();
+            }
+            else if (mapping.TypeDesc.FormatterName == "String")
+            {
+                if (mapping.TypeDesc.CollapseWhitespace)
+                {
+                    return CollapseWhitespace(readFunc());
+                }
+                else
+                {
+                    return readFunc();
+                }
+            }
+            else
+            {
+                string methodName = $"To{mapping.TypeDesc.FormatterName}";
+                if (!mapping.TypeDesc.HasCustomFormatter)
+                {
+                    MethodInfo method = typeof(XmlConvert).GetMethod(methodName, BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, new Type[] { typeof(string) });
+                    return method.Invoke(null, new object[] { readFunc() });
+                }
+                else
+                {
+                        MethodInfo method = typeof(XmlSerializationReader).GetMethod(methodName, BindingFlags.Static | BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, new Type[] { typeof(string) });
+                        return method.Invoke(this, new object[] { readFunc() });
+                }
+            }
+        }
+
+        object WriteStructMethod(StructMapping mapping, bool isNullable, bool checkType, string defaultNamespace)
+        {
+            if (mapping.IsSoap)
+                throw new PlatformNotSupportedException();
```

Comment:
>throw new PlatformNotSupportedException(); [](start = 16, length = 42)

Inconsistent exception message. On line 639 you are throwing an exception with "mapping.IsSoap" as the message.